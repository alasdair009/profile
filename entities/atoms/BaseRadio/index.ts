export { BaseRadio } from "./BaseRadio";
export type { BaseRadioProps } from "./BaseRadio";
