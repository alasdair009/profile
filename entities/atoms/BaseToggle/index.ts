export { BaseToggle } from "./BaseToggle";
export type { BaseToggleProps } from "./BaseToggle";
