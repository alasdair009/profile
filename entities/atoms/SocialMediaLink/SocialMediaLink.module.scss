@use "../../../app/styles/animation";
@use "../../../app/styles/colors";
@use "../../../app/styles/dimensions";
@use "../../../app/styles/helpers/rem";
@use "sass:map";

$socialMediaIcons: (
  bluesky: url(./assets/bluesky.svg),
  facebook: url(./assets/facebook.svg),
  github: url(./assets/github.svg),
  instagram: url(./assets/instagram.svg),
  linkedin: url(./assets/linkedin.svg),
  x: url(./assets/x.svg),
  youtube: url(./assets/youtube.svg),
);

.root {
  align-items: center;
  aspect-ratio: 1;
  background: colors.$colorblackEvil;
  color: transparent;
  display: flex;
  justify-content: center;
  -webkit-mask: no-repeat center;
  mask: no-repeat center;
  -webkit-mask-size: contain;
  mask-size: contain;
  overflow: hidden;
  position: relative;
  width: rem.toRem(map.get(dimensions.$sizes, s48));

  &::before {
    background: linear-gradient(
        colors.$colorWhiteGhost 50%,
        colors.$colorGreenGrass 50%,
        colors.$colorGreenGrass 100%
      )
      0 0;
    content: "";
    height: 200%;
    left: 0;
    position: absolute;
    top: 0;
    transition: all animation.animationDuration(0.25);
    width: 100%;
  }

  &:hover::before {
    top: -100%;
  }
}
