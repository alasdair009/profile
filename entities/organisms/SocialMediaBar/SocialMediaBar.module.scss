@use "../../../app/styles/dimensions";
@use "../../../app/styles/helpers/rem";

.root {
  align-items: center;
  display: flex;
  gap: dimensions.getSize(s8);
  justify-content: center;
  margin: 0 auto;
  max-width: rem.toRem(dimensions.$globalContentMaxWidth);
  width: 100%;
}
