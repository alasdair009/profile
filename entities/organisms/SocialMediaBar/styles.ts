"use client"

import styled from "styled-components";
import {globalContentMaxWidth, sizes} from "@/entities";
import {rem} from "polished";

export const Root = styled.nav`
  align-items: center;
  display: flex;
  gap: ${sizes.s8.rem};
  justify-content: center;
  margin: 0 auto;
  max-width: ${rem(globalContentMaxWidth)};
  width: 100%;
`;