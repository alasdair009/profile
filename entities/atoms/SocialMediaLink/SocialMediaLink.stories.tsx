import { SocialMediaLink } from "./SocialMediaLink";
import { Meta, StoryObj } from "@storybook/react";
import { expect, userEvent, within } from "@storybook/test";

const meta: Meta<typeof SocialMediaLink> = {
  component: SocialMediaLink,
};
export default meta;

export const X: StoryObj<typeof SocialMediaLink> = {
  args: {
    variant: "x",
  },
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const socialMediaElement = canvas.getByTestId(SocialMediaLink.name);

    await expect(socialMediaElement).toBeInTheDocument();
    await expect(socialMediaElement).toHaveAttribute(
      "href",
      "https://twitter.com/alasdair009"
    );
  },
};

export const LinkedIn: StoryObj<typeof SocialMediaLink> = {
  args: {
    variant: "linkedin",
  },
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const socialMediaElement = canvas.getByTestId(SocialMediaLink.name);

    await expect(socialMediaElement).toBeInTheDocument();
    await expect(socialMediaElement).toHaveAttribute(
      "href",
      "https://www.linkedin.com/in/alasdairmacrae/"
    );
  },
};

export const GitHub: StoryObj<typeof SocialMediaLink> = {
  args: {
    variant: "github",
  },
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const socialMediaElement = canvas.getByTestId(SocialMediaLink.name);

    await expect(socialMediaElement).toBeInTheDocument();
    await expect(socialMediaElement).toHaveAttribute(
      "href",
      "https://github.com/alasdair009"
    );
  },
};

export const Facebook: StoryObj<typeof SocialMediaLink> = {
  args: {
    variant: "facebook",
  },
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const socialMediaElement = canvas.getByTestId(SocialMediaLink.name);

    await expect(socialMediaElement).toBeInTheDocument();
    await expect(socialMediaElement).toHaveAttribute(
      "href",
      "https://www.facebook.com/alasdair009"
    );
  },
};

export const Instagram: StoryObj<typeof SocialMediaLink> = {
  args: {
    variant: "instagram",
  },
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const socialMediaElement = canvas.getByTestId(SocialMediaLink.name);

    await expect(socialMediaElement).toBeInTheDocument();
    await expect(socialMediaElement).toHaveAttribute(
      "href",
      "https://www.instagram.com/alasdair009"
    );
  },
};

export const YouTube: StoryObj<typeof SocialMediaLink> = {
  args: {
    variant: "youtube",
  },
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const socialMediaElement = canvas.getByTestId(SocialMediaLink.name);

    await expect(socialMediaElement).toBeInTheDocument();
    await expect(socialMediaElement).toHaveAttribute(
      "href",
      "https://www.youtube.com/alasdair009"
    );
  },
};
