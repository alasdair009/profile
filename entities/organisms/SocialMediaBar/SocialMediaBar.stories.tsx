import { SocialMediaBar } from "./SocialMediaBar";
import { Meta, StoryObj } from "@storybook/nextjs";
import { expect, within } from "storybook/test";
import { SocialMediaLink } from "../../atoms/SocialMediaLink";

const meta: Meta<typeof SocialMediaBar> = {
  component: SocialMediaBar,
  parameters: {
    controls: { hideNoControlsWarning: true },
  },
  subcomponents: {
    // @ts-ignore
    SocialMediaLink: SocialMediaLink,
  },
};
export default meta;

export const Default: StoryObj<typeof SocialMediaBar> = {
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const socialMediaBarElement = canvas.getByTestId(SocialMediaBar.name);
    const socialLinks = canvas.getAllByTestId(SocialMediaLink.name);

    await expect(socialMediaBarElement).toBeInTheDocument();
    await expect(socialLinks).toHaveLength(7);
  },
};
