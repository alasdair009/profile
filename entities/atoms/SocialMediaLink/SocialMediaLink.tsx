import { Root } from "./styles";
import { HTMLAttributes } from "react";
import { SocialMediaLinkVariant } from "./SocialMediaLinks.types";

type SocialMediaProps = {
  /**
   * Network to link to.
   */
  variant: SocialMediaLinkVariant;
} & HTMLAttributes<HTMLAnchorElement>;

const getSocialMediaUrl = (variant: SocialMediaLinkVariant) => {
  switch (variant) {
    case "github":
      return "https://github.com/alasdair009";
    case "linkedin":
      return "https://www.linkedin.com/in/alasdairmacrae/";
    case "x":
      return "https://twitter.com/alasdair009";
  }
};

/**
 * Icon that links to the requested social media network.
 */
export function SocialMediaLink({ variant, children, ...rest }: SocialMediaProps) {
  return (
    <Root href={getSocialMediaUrl(variant)} target="_blank" rel="noopener" $variant={variant} {...rest}>
      {children}
    </Root>
  );
}
