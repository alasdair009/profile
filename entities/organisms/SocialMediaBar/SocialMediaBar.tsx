import { globalContentMaxWidth, sizes, SocialMediaLink } from "@/entities";
import { Root } from "./styles";
import { HTMLAttributes } from "react";

type SocialMediaBarProps = HTMLAttributes<HTMLDivElement>;
export function SocialMediaBar({ ...rest }: SocialMediaBarProps) {
  return (
    <Root {...rest}>
      <SocialMediaLink variant="x">X</SocialMediaLink>
      <SocialMediaLink variant="linkedin">LinkedIn</SocialMediaLink>
    </Root>
  );
}
