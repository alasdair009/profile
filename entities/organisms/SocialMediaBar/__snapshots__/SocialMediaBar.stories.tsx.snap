// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`organisms/SocialMediaBar Default smoke-test 1`] = `
<nav class="sc-hDcvty hXbJQq">
  <a target="_blank"
     rel="noopener"
     data-testid="SocialMediaLink"
     class="sc-euGpHm iBLaxx"
     href="https://twitter.com/alasdair009"
  >
    X
  </a>
  <a target="_blank"
     rel="noopener"
     data-testid="SocialMediaLink"
     class="sc-euGpHm joEhIl"
     href="https://www.linkedin.com/in/alasdairmacrae/"
  >
    LinkedIn
  </a>
  <a target="_blank"
     rel="noopener"
     data-testid="SocialMediaLink"
     class="sc-euGpHm hKfaBN"
     href="https://github.com/alasdair009"
  >
    GitHub
  </a>
</nav>
`;
